streaming/vision/convert/coco.py

```diff
@@ -0,0 +1,231 @@
+            self.label_info[cnt] = cat['name']
+
+        # build inference for images
+        for img in self.data['images']:
+            img_id = img['id']
+            img_name = img['file_name']
+            img_size = (img['height'], img['width'])
+            if img_id in self.images:
+                raise Exception('dulpicated image record')
+            self.images[img_id] = (img_name, img_size, [])
+
+        # read bboxes
+        for bboxes in self.data['annotations']:
+            img_id = bboxes['image_id']
+            bbox = bboxes['bbox']
+            bbox_label = self.label_map[bboxes['category_id']]
+            self.images[img_id][2].append((bbox, bbox_label))
+
+        for k, v in list(self.images.items()):
+            if len(v[2]) == 0:
+                self.images.pop(k)
+
+        self.img_keys = list(self.images.keys())
+
+    #@property
+    def labelnum(self):
+        return len(self.label_info)
+
+    def __len__(self):
+        return len(self.images)
+
+    def __getitem__(self, idx: int):
+        img_id = self.img_keys[idx]
+        img_data = self.images[img_id]
+        fn = img_data[0]
+        img_path = os.path.join(self.img_folder, fn)
+
+        img = Image.open(img_path).convert('RGB')
+
+        htot, wtot = img_data[1]
+        bbox_sizes = []
+        bbox_labels = []
+
+        for (l, t, w, h), bbox_label in img_data[2]:
+            r = l + w
+            b = t + h
+            bbox_size = (l / wtot, t / htot, r / wtot, b / htot)
+            bbox_sizes.append(bbox_size)
+            bbox_labels.append(bbox_label)
+
+        bbox_sizes = torch.tensor(bbox_sizes)
+        bbox_labels = torch.tensor(bbox_labels)
+
+        return img, img_id, (htot, wtot), bbox_sizes, bbox_labels
+
+
+def parse_args() -> Namespace:
+    """Parse command line arguments.
+
+    Args:
+        Namespace: Command line arguments.
+    """
+    args = ArgumentParser()
+    args.add_argument('--in_root',
+                      type=str,
+                      default='./datasets/coco/',
+                      help='Location of Input dataset. Default: ./datasets/coco/')
+    args.add_argument(
+        '--out_root',
+        type=str,
+        default='./datasets/mds/coco/',
+        help='Location to store the compressed dataset. Default: ./datasets/mds/coco/')
+    args.add_argument('--splits',
+                      type=str,
+                      default='train,val',
+                      help='Split to use. Default: train,val')
+    args.add_argument('--compression',
+                      type=str,
+                      default='zstd:7',
+                      help='Compression algorithm to use. Default: zstd:7')
+    args.add_argument('--hashes',
+                      type=str,
+                      default='sha1,xxh64',
+                      help='Hashing algorithms to apply to shard files. Default: sha1,xxh64')
+    args.add_argument(
+        '--limit',
+        type=int,
+        default=1 << 25,
```

Comment:
please double check shard size limits: because we intend to train this on 8xA100 with 8 workers/process, number of resulting shards should preferably be comfortably above 64

Reply:
The number of shards is way higher than 64 for COCO so we need to change anything here. Thanks!